### PR TITLE
add some fixes to SizeLimitHandler

### PR DESCRIPTION
**Pull request by lachlan-roberts**
Signed-off-by: Lachlan Roberts <lachlan@webtide.com>

### CODE REVIEW

File: jetty-server/src/main/java/org/eclipse/jetty/server/handler/SizeLimitHandler.java [hunk 1]
```diff
@@ -126,6 +126,9 @@ public boolean isOptimizedForDirectBuffers()
         @Override
         public HttpInput.Content readFrom(HttpInput.Content content)
         {
+            if (content == null)
+                return null;
```
Comment (lachlan-roberts) [inline jetty-server/src/main/java/org/eclipse/jetty/server/handler/SizeLimitHandler.java:130]: maybe this should also be resetting `_read` back to 0.

**Comment by joakime** [comment]
Regarding the AllowedResourceAliasCheckTest failures, I'll need to backport (to `jetty-9.4.x`) the fixes from PR #8872

**Comment by joakime** [comment]
Opened #8875 for issue in `AllowedResourceAliasCheckTest` (it's just a backport of a previous stability fix from `jetty-10.0.x`)

File: jetty-server/src/main/java/org/eclipse/jetty/server/handler/SizeLimitHandler.java (hunk 1 above)
Comment (gregw) [reply jetty-server/src/main/java/org/eclipse/jetty/server/handler/SizeLimitHandler.java:130]: I don't think so, as reading null just means there is nothing currently available, but there may be after a demand.

**Review by gregw — APPROVED** [review]

**Pull request merged**